armi/reactor/tests/test_blocks.py

```diff
@@ -2270,12 +2275,96 @@ def test_gridNotCreatedMultipleMultiplicities(self):
         }
         # add a wire only some places in the block, so grid should not be created.
         wire = components.Helix("wire", "HT9", **wireDims)
-        self.HexBlock.add(wire)
-        self.HexBlock.spatialGrid = None  # clear existing
+        self.hexBlock.add(wire)
+        self.hexBlock.spatialGrid = None  # clear existing
         with self.assertRaises(ValueError):
-            self.HexBlock.autoCreateSpatialGrids()
+            self.hexBlock.autoCreateSpatialGrids()
+
+        self.assertIsNone(self.hexBlock.spatialGrid)
+
+
+class TestHexBlockOrientation(unittest.TestCase):
+    def setUp(self):
+        self.td = TemporaryDirectoryChanger()
+        self.td.__enter__()
+
+    def tearDown(self):
+        self.td.__exit__(None, None, None)
+
+    @staticmethod
+    def getLocalCoordinatesBlockBounds(b: blocks.HexBlock):
+        """Call getLocalCoordinates() for every Component in the Block and find the X/Y bounds."""
+        maxX = -111
+        minX = 999
+        maxY = -111
+        minY = 999
+        for comp in b:
+            locs = comp.spatialLocator
+            if not isinstance(locs, grids.MultiIndexLocation):
+                locs = [locs]
+
+            for loc in locs:
+                x, y, _ = loc.getLocalCoordinates()
+                if x > maxX:
+                    maxX = x
+                elif x < minX:
+                    minX = x
+
+                if y > maxY:
+                    maxY = y
+                elif y < minY:
+                    minY = y
+
+        return minX, maxX, minY, maxY
+
+    def test_validateReactorCornersUp(self):
+        """Validate the spatial grid for a corners up HexBlock and its children."""
+        # load a corners up reactor
+        _o, r = loadTestReactor(
+            os.path.join(TEST_ROOT, "smallestTestReactor"),
+            inputFileName="armiRunSmallest.yaml",
+        )
+
+        # grab a pinned fuel block, and verify it is corners up
+        b = r.core.getFirstBlock(Flags.FUEL)
+        self.assertTrue(r.core.spatialGrid.cornersUp)
+        self.assertFalse(b.spatialGrid.cornersUp)
+        self.assertNotEqual(r.core.spatialGrid.cornersUp, b.spatialGrid.cornersUp)
+
+        # for a flats up block-grid, the hex centroids should stretch more in X than Y
+        minX, maxX, minY, maxY = self.getLocalCoordinatesBlockBounds(b)
+        ratio = (maxY - minY) / (maxX - minX)
+        self.assertAlmostEqual(ratio, math.sqrt(3) / 2, delta=0.0001)
```

Comment:
I think shows we have a pin grid that is of the opposite orientation we want. The pins on a block in a corners up reactor should be on a flats up lattice. However, this test shows that while `b.spatialGrid` indicates a flats up grid, the actual pins are located on a corners up lattice.

1. If the block resides in a corners up grid, it's pin should be in a flats up orientation.
2. A hexagonal lattice in the flats up orientation stretches more in Y than in X.
   - The maximum Y distance between adjacent lattice sites is the pitch. Sites stack in the y-direction.
   - The X distance between adjacent lattice sites is `pitch * cos(30) = p * sqrt(3) / 2 < pitch`
   - For a single ring, `maxY - minY = 2 * pitch`
   - For a single ring, `maxX - minX = 2 * pitch * sqrt(3) / 2 = pitch * sqrt(3)`
3. The ratio should be `2 * pitch / (pitch * sqrt(3)) = 2 / sqrt(3)`
4. A corners up lattice would be and stretch longer in X than in Y, and have a ratio of `sqrt(3) / 2` like is tested here.